DecodeCx1.py

```diff
@@ -5,195 +5,160 @@
-    'workspace_id': '01GG7AW75CHDNZ2BNYKZS9ZHMF'}
-
-end_points1='tester'
-end_points2=['test/blocks?study_id=$study_id&block_id=$block_id&tester_id=$tester','tester_response']
-end_points_2=Template(end_points2[0])
-end_points3=['studies/4def2733-f775-462b-a65f-2d20469064fe?is_insights=true','testers','testers',f'blocks?study_id={study_id}&block_id=99d22693-ed4f-4474-b42a-ffa89a519b49',f'blocks?study_id={study_id}&block_id=5a43eaba-8700-4524-9eb5-1fb09bba1e15','blocks_insights']
-
-for  t in range(2):
-
-    block_id='99d22693-ed4f-4474-b42a-ffa89a519b49'
-    option_ids=[]
-    tester=''
-
-
-#To generate tester id using post method******************************************************************************************************
-    def call_tester_url(a):
-
-        global tester_details
-        global block_id
-        global option_ids
-        global tester
-        global r
-        
-        url=a
+login_url='https://app.dev.getdecode.io/authentication/login'
+tester_url='tester'
+get_resp_url='test/blocks?study_id=$study_id&block_id=$block_id&tester_id=$tester'
+get_resp_url_2=Template(get_resp_url)
+post_resp_url='tester_response'
+insights_url='blocks_insights'
+
+def generate_headers():
+#To generater authorization id and id token***********************************************************************************************************
+    request_json={'username': "bantupalli.sriharsha@entropiktech.com", 'password': "Harsha@10150", 'workspace': "beintern"}
+    jsondata=json.dumps(request_json)
+    response=requests.post(login_url, data=jsondata)
+    assert response.status_code==200, 'Error'
+    c=response.content
+    c=json.loads(c.decode('utf-8'))
+    access_token=c['data']['access_token']
+    id_token=c['data']['id_token']
+    return {'authorization': access_token, 'id_token': id_token, 'workspace_id': '01GG7AW75CHDNZ2BNYKZS9ZHMF'}
+
+
+
+def generate_testers(url):
+#To generate tester id using post method*********************************************************************************************************

```

Comment:
maintain the indentation to this comment, and remove *'s if there is no advantage with it.

Reply:
sure..